### PR TITLE
Several enhancements for optimization

### PR DESCRIPTION
**Pull request by emailweixu**
1. Parameter specific weight decay for AdamTF
2. Bias only option for BatchNorm
3. Support fixed_norm for optimizer
4. Put optimizer hyper parameters to optimizer.defaults so that their values can be shown in tensorboard summary of optimizers

### CODE REVIEW

File: alf/batch_norm.py [hunk 1]
```diff
@@ -38,7 +41,19 @@ def __init__(self,
         self._track_running_stats = track_running_stats
         if affine:
             self._weight = nn.Parameter(torch.Tensor(num_features))
+            use_bias = True
+        else:
+            self._weight = None
+        if use_bias:
+            if self._weight is None:
+                # pytorch has a bug which cannot handle the case that weihgt is
```
Comment (Haichao-Zhang) [inline alf/batch_norm.py]: weihgt -> weight

File: alf/batch_norm.py [hunk 2]
```diff
@@ -102,8 +117,9 @@ def reset_parameters(self):
                 self._running_means[i].zero_()
                 self._running_vars[i].fill_(1)
                 self._num_batches_tracked[i].zero_()
-        if self._affine:
+        if self._weight is not None:
```
Comment (Haichao-Zhang) [inline alf/batch_norm.py:120]: We can perhaps skip initializing for _weight in the bias only case?

File: alf/optimizers/optimizers.py [hunk 3]
```diff
@@ -188,6 +200,10 @@ def step(self, closure=None):
 
         super(NewCls, self).step(closure=closure)
 
+        for param in params:
+            if param.grad is not None and getattr(param, 'fixed_norm', False):
+                param.data.mul_(self._norms[param] / (param.norm() + 1e-30))
```
Comment (Haichao-Zhang) [inline alf/optimizers/optimizers.py:216]: It's not quite clear what the operation this ``fixed_norm` option represents. Maybe we can add some explanations.

File: alf/networks/projection_networks.py [hunk 4]
```diff
@@ -34,6 +34,8 @@ def __init__(self,
                  input_size,
                  action_spec,
                  logits_init_output_factor=0.1,
+                 weight_opt_args=None,
+                 bias_opt_args=None,
```
Comment (hnyu) [inline alf/networks/projection_networks.py:38]: Why are these arguments only added for this class but not for the other ones?

File: alf/batch_norm.py (hunk 1 above)
Comment (emailweixu) [reply alf/batch_norm.py]: fixed.

File: alf/batch_norm.py (hunk 2 above)
Comment (emailweixu) [reply alf/batch_norm.py:120]: It does not matter since it's one

File: alf/networks/projection_networks.py (hunk 4 above)
Comment (emailweixu) [reply alf/networks/projection_networks.py:38]: No particular reason. It's just that I didn't use the others in experiments.

**Review by hnyu — APPROVED** [review]
